linera-service/src/cli_wrappers/wallet.rs

```diff
@@ -450,9 +452,17 @@ impl ClientWrapper {
         if let Ok(var) = env::var(CLIENT_SERVICE_ENV) {
             command.args(var.split_whitespace());
         }
-        let child = command
-            .args(["--port".to_string(), port.to_string()])
-            .spawn_into()?;
+        if let Some(requested_port) = port.into() {
+            command.args(["--port".to_string(), requested_port.to_string()]);
+        }
+        let mut child = command.spawn_into()?;
+        let port = Self::parse_node_service_port(
+            child
+                .stdout
+                .take()
```

Comment:
The downside of this is that the node service logs are no longer printed with the test output. Not sure what should be done here. One option is to spawn a task that keeps reading it and printing it out with `eprintln` :thinking:

Reply:
All machine-readable results must go to stdout.